### PR TITLE
filters: more tests to node/cn/filters/filter_test.go

### PR DESCRIPTION
**Pull request by ehnuje**
## Proposed changes

- Added `node/cn/filters/filter_test.go` to increase the coverage of `filter` package
- Removed unnecessary `db` from `Filter` struct

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: node/cn/filters/filter_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	topic1 := addr1.Hash()
+	topic2 := addr2.Hash()
+	topics := [][]common.Hash{{topic1}, {topic2}}
+
+	begin := int64(12345)
+	end := int64(54321)
+
+	newFilter := New(mockBackend, begin, end, addrs, topics)
+	assert.NotNil(t, newFilter)
+	assert.Equal(t, begin, newFilter.begin)
```
Comment (KimKyungup) [inline node/cn/filters/filter_test.go]:
```suggestion
	assert.Equal(t, mockBackend, newFilter.backend)
	assert.Equal(t, begin, newFilter.begin)
```

File: node/cn/filters/filter_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	assert.Equal(t, end, newFilter.end)
+	assert.Equal(t, topics, newFilter.topics)
+	assert.Equal(t, addrs, newFilter.addresses)
+}
```
Comment (KimKyungup) [inline node/cn/filters/filter_test.go]:
```suggestion
	assert.NotNil(t, newFilter.matcher)
}
```

**Comment by ehnuje** [comment]
@jeonghonah @kjhman21 @nogang Dear reviewers, could you please review this change?

File: node/cn/filters/filter_test.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	assert.Equal(t, mockBackend, newFilter.backend)
+	assert.Equal(t, begin, newFilter.begin)
+	assert.Equal(t, end, newFilter.end)
+	assert.Equal(t, topics, newFilter.topics)
+	assert.Equal(t, addrs, newFilter.addresses)
+	assert.NotNil(t, newFilter.matcher)
+}
+
+func TestFilter_Logs(t *testing.T) {
+
```
Comment (kjhman21) [inline node/cn/filters/filter_test.go]: Is this function intentionally empty?

Comment (ehnuje) [reply node/cn/filters/filter_test.go]: No, it is my mistake. I will update the PR, Thanks!

**Comment by ehnuje** [comment]
@jeonghonah @nogang @kjhman21 @KimKyungup Dear reviewers, could you please review this?

File: node/cn/filters/filter_test.go [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+	r := &types.Receipt{GasUsed: cumulativeGasUsed}
+	if failed {
+		r.Status = types.ReceiptStatusFailed
+	} else {
+		r.Status = types.ReceiptStatusSuccessful
+	}
+	return r
+}
+
+func TestFilters(t *testing.T) {
```
Comment (ehnuje) [inline node/cn/filters/filter_test.go]: Imported this test from go-ethereum.

File: node/cn/filters/filter.go [hunk 5]
```diff
@@ -52,17 +53,16 @@ type Backend interface {
...
 
 	matcher *bloombits.Matcher
 }
 
-// New creates a new filter which uses a bloom filter on blocks to figure out whether
-// a particular block is interesting or not.
-func New(backend Backend, begin, end int64, addresses []common.Address, topics [][]common.Hash) *Filter {
+// NewRangeFilter creates a new filter which uses a bloom filter on blocks to
+// figure out whether a particular block is interesting or not.
+func NewRangeFilter(backend Backend, begin, end int64, addresses []common.Address, topics [][]common.Hash) *Filter {
```
Comment (ehnuje) [inline node/cn/filters/filter.go]: Change the function name in accordance with the name in go-ethereum.

**Comment by ehnuje** [comment]
@jeonghonah @nogang @kjhman21 Dear reviewers, could you please review this change?

**Pull request merged**